### ISSUE TITLE
Bag.to_dataframe().compute() returns empty tuple if bag is empty

### ISSUE THREAD

**Issue by ian-whitestone** [dask/dask#4321]
Similar to what was discussed in #2884, there are situations that arise where a bag may be empty due to filtering, or building a bag from a path where there aren't any files. I have a job that runs daily that does the following:

1) Builds a dask bag from a bunch of (date partitioned) files in S3
2) Filters the bag based on some criteria
3) Converts it to a dask dataframe (based on discussions in #2884, i provide the column types since the bag may become empty due to filtering)
4) Converts it to a pandas dataframe
5) Does other stuff (like join to other dataframes created through the same process outlined in steps 1-4)

When a bag becomes empty due to filtering, I get the expected result (an empty pandas dataframe) at the end of step 4.

```python
>>> import dask.bag

# Expected result
>>> b = dask.bag.from_sequence([{'a':1}, {'a':2}, {'a':3}, {'a':4}])
>>> b = b.filter(lambda x: x['a'] == 0)
>>> dd = b.to_dataframe(meta={'a': 'int'})
>>> dd.compute()
Empty DataFrame
Columns: [a]
Index: []
```

When a bag is empty because there are no files in the path, I get an unexpected result (an empty tuple) at the end of step 4.

```python
# Unexpected result
>>> b = dask.bag.from_sequence([])
>>> b = b.filter(lambda x: x['a'] == 0)
>>> dd = b.to_dataframe(meta={'a': 'int'})
>>> dd.compute()
()
```

Since I am supplying the column metadata, I would expect an empty pandas dataframe. What are your thoughts?

**Versions:**

Python 3.6.5
Dask 0.19.3
Pandas 0.23.4

**Comment by mrocklin** [comment]
Reproduced.  Thank you for the excellent bug report @ian-whitestone .  I agree with you that this should  produce an empty pandas dataframe.

It looks like the task graph and the keys are both empty, which is a bit surprising.

```pythonIn [13]: dict(dd.dask)
Out[13]: {}

In [14]: dd.__dask_keys__()
Out[14]: []
```

It may be that we can handle this entirely on the dask dataframe side.  It should probably know how to construct itself in the event that it has zero partitions.

It looks like dask.bag does the same.  Perhaps `dask.bag.from_sequence` should always include at least a single partition, even if that partition is empty.

This is a moderately doable task for someone lightly familiar with Dask internals if anyone wants to take it on.

**Comment by jrbourbeau** [comment]
Closing as this was fixed in #4475.
